messaging/src/main/java/org/jboss/as/messaging/TransportConfigOperationHandlers.java

```diff
@@ -145,6 +145,16 @@ static void processConnectors(final OperationContext context, final Configuratio
                 final ModelNode config = property.getValue();
                 final Map<String, Object> parameters = getParameters(context, config);
                 final String clazz = FACTORY_CLASS.resolveModelAttribute(context, config).asString();
+                if (NettyConnectorFactory.class.getName().equals(clazz)) {
```

Comment:
Regardless of the value of the factory-class, the generic `connector` resource defines the `socket-binding` attribute, so you can put it in any case (if it is defined though)